xprocspec-runner/src/main/java/org/daisy/maven/xproc/xprocspec/XProcSpecRunner.java

```diff
@@ -85,6 +88,27 @@ public boolean run(Map<String,File> tests,
 			activate();
 		engine.setCatalog(catalog);
 		
+		// register Java implementation of px:message
+		// FIXME: make a generic step that can be used by all XProc engines
+		if (engine.getClass().getName().equals("org.daisy.maven.xproc.calabash.Calabash")) {
+			try {
+				engine.getClass().getMethod("setDefaultConfiguration", Reader.class).invoke(engine, new StringReader(
```

Comment:
So IIUC you set the configuration with reflection. Isn't there a risk of losing whatever is relevant in the engine's default configuration?